File: apzkr-pzpi-21-2-butsulin-ihor/Task4-MobileClient/APZMobile/app/src/main/java/com/osigram/apz2024/mobile/models/Worker.kt
package com.osigram.apz2024.mobile.models

import kotlinx.serialization.Serializable

@Serializable
data class Worker(val ID: ULong, val userID: ULong, val user: User, val warehouseID: ULong, val timetables: List<Timetable>?)

File: apzkr-pzpi-21-2-butsulin-ihor/Task1-Server/apz-backend/types/models/Worker.go
package models

import (
	"gorm.io/gorm"
	"time"
)

type Worker struct {
	gorm.Model
	UserID      uint        `json:"userID"`
	User        User        `json:"user"`
	WarehouseID uint        `json:"warehouseID"`
	Timetables  []Timetable `json:"timetables"`
}

func (w Worker) IsWorking() bool {
	for _, timetable := range w.Timetables {
		if timetable.StartTime.Before(time.Now()) && timetable.EndTime.After(time.Now()) {
			return true
		}
	}
	return false
}
